orchestrator-salt/src/main/resources/salt/salt/cloudera/manager/scripts/generate_agent_tokens.sh.j2

```diff
@@ -35,7 +35,7 @@ do
   if [[ ! -e ${tokendir}/agent_token_copied_to_client ]]; then
     set +e
     # This can be fairly slow, for large clusters
-    COPY_RESULT=$(salt ${fqdn} cp.get_file salt://agent-tls-tokens/${AGENT_FQDN_TOKENDIR_MAP[${fqdn}]}/cmagent.token /etc/cloudera-scm-agent/cmagent.token)
+    COPY_RESULT=$(salt ${fqdn} cp.get_file salt://agent-tls-tokens/${AGENT_FQDN_TOKENDIR_MAP[${fqdn}]}/cmagent.token /etc/cloudera-scm-agent/cmagent.token makedirs=True)
```

Comment:
what about the rights of the created directory? would it be the same as when it's created during installation?

Reply:
I've created 4 cluster with it, but let me check.

Reply:
A naive idea: Why not add a `require: - cmd: install-cloudera-manager-server` dependency to state `run_generate_agent_tokens`?

Reply:
this runs on the CM server instance, agent is installed everywhere. you can't wait for every instance to finish a state in saltstack (not strictly true, but the way we use it we can't enforce it)

Reply:
Checked on a fresh installment: 

On idbroker1 after a green provision, I removed the yum cloudera-manager-agent packed, remove the directory and reinstalled it with yum.
Compred to other nodes, it has the same user and group.

`
lnardai-test-dh4-broker1.lnardai.xcu2-8y8x.wl.cloudera.site:
    drwxr-xr-x    2 root     root         24 Aug  5 13:39 cloudera-scm-agent
lnardai-test-dh4-broker0.lnardai.xcu2-8y8x.wl.cloudera.site:
    drwxr-xr-x    2 root     root         92 Aug  3 16:03 cloudera-scm-agent
lnardai-test-dh4-broker2.lnardai.xcu2-8y8x.wl.cloudera.site:
    drwxr-xr-x    2 root     root         92 Aug  3 16:03 cloudera-scm-agent
`